src/context/VideoContext/VideoContext.jsx

```diff
@@ -0,0 +1,32 @@
+import {createContext, useContext, useEffect, useState} from "react"
+import axios from "axios"
+// create context
+
+const VideoContext = createContext({})
+
+//provide context
+
+const VideoProvider = ({children})=>{
+    const [videosData, setVideosData] = useState([])
+    const fetchVideo = async ()=>{
+        try{
+            const {data} = await axios.get('/api/videos')
+            setVideosData(data.videos)
+        }
+        catch(error){
+            console.log("Error in fetching data")
+        }
+    }
```

Comment:
You can define this function inside `useEffect`, then there will be no warnings for missing dependency

Reply:
warning is not coming anymore